### PR TITLE
Add selenium test case for show and hide title of paragraph

### PR DESCRIPTION
**Pull request by ravicodder**
What is this PR for?

Add a new test case for testing the show title and hide title button.

What type of PR is it?

Test

Is there a relevant Jira issue?

NA

How should this be tested?

On OSX, you'll need firefox 42.0 installed, then you can run with

PATH=~/Applications/Firefox.app/Contents/MacOS/:$PATH CI="" \
mvn -Dtest=org.apache.zeppelin.ParagraphActionsIT -Denforcer.skip=true \
test -pl zeppelin-server

Questions:

Does the licenses files need update?NO
Is there breaking changes for older versions?NO
Does this needs documentation?NO

### CODE REVIEW

**Comment by felixcheung** [comment]
@prabhjyotsingh could you review this as well?

**Comment by prabhjyotsingh** [comment]
@ravicodder  can you please rebase again.

File: zeppelin-server/src/test/java/org/apache/zeppelin/integration/ParagraphActionsIT.java [hunk 1]
```diff
@@ -120,6 +120,61 @@ public void testMoveUpAndDown() throws Exception {
...
+              CoreMatchers.equalTo("NEW TITLE"));
+      ZeppelinITUtils.sleep(1000,false);
+      driver.navigate().refresh();
+      collector.checkThat("After Page Refresh : The title field contains ",
+              driver.findElement(By.xpath(getParagraphXPath(1) + "//div[contains(@class, 'binding')]")).getText(),
+              CoreMatchers.equalTo("NEW TITLE"));
+      deleteTestNotebook(driver);
+
+    } catch (ElementNotVisibleException e) {
+      LOG.error("Exception in ParagraphActionsIT while testTitleButton ", e);
```
Comment (prabhjyotsingh) [inline zeppelin-server/src/test/java/org/apache/zeppelin/integration/ParagraphActionsIT.java]: can you do Exception instead of ElementNotVisibleException

**Comment by felixcheung** [comment]
Error:
```
09:55:34,147 ERROR org.apache.zeppelin.integration.ZeppelinIT:257 - Exception in ZeppelinIT while testSparkInterpreterDependencyLoading 
org.openqa.selenium.TimeoutException: Timed out after 60 seconds waiting for org.apache.zeppelin.AbstractZeppelinIT$1@3d56054b
```
Could you check?

File: zeppelin-server/src/test/java/org/apache/zeppelin/integration/ParagraphActionsIT.java [hunk 2]
```diff
@@ -120,6 +120,61 @@ public void testMoveUpAndDown() throws Exception {
...
+      driver.navigate().refresh();
+      collector.checkThat("After Page Refresh : The title field contains ",
+              driver.findElement(By.xpath(getParagraphXPath(1) + "//div[contains(@class, 'binding')]")).getText(),
+              CoreMatchers.equalTo("NEW TITLE"));
+      deleteTestNotebook(driver);
+
+    } catch (Exception e) {
+      LOG.error("Exception in ParagraphActionsIT while testTitleButton ", e);
+      File scrFile = ((TakesScreenshot) driver).getScreenshotAs(OutputType.FILE);
+      throw e;
```
Comment (prabhjyotsingh) [inline zeppelin-server/src/test/java/org/apache/zeppelin/integration/ParagraphActionsIT.java]: throw e (Exception);  but signature throws InterruptedException, can you change that ?

**Comment by ravicodder** [comment]
@felixcheung 
@prabhjyotsingh 
Used handle exception please have a look

**Comment by prabhjyotsingh** [comment]
LGTM

File: zeppelin-server/src/test/java/org/apache/zeppelin/integration/ParagraphActionsIT.java [hunk 3]
```diff
@@ -233,6 +233,60 @@ public void testMoveUpAndDown() throws Exception {
...
+              driver.findElement(By.xpath(getParagraphXPath(1) + "//a[contains(@ng-click, 'showTitle()')]")).getText(),
+              CoreMatchers.equalTo("Show title"));
+
+      driver.findElement(By.xpath(getParagraphXPath(1) + "//ul/li/a[@ng-click='showTitle()']")).click();
+      collector.checkThat("After Show Title : The title field contains",
+              driver.findElement(By.xpath(getParagraphXPath(1) + "//div[contains(@class, 'binding')]")).getText(),
+              CoreMatchers.equalTo("Untitled"));
+
+      driver.findElement(By.xpath(getParagraphXPath(1) + "//span[@class='icon-settings']")).click();
+      collector.checkThat("After Show Title : The title option in option panel of paragraph is labeled as",
```
Comment (corneadoug) [inline zeppelin-server/src/test/java/org/apache/zeppelin/integration/ParagraphActionsIT.java]: After Show Title -> After Hide Title

**Comment by corneadoug** [comment]
When you use multiple times the same XPath, it would be better to have a reusable variable for it.

This Xpath for the title `//div[contains(@class, 'binding')]` might not be precise enough. This is targeting the title a bit more: `//div[contains(@class, 'title')]/div`

**Comment by ravicodder** [comment]
@corneadoug Thank you for pointing it out. Changes done, please have a look.

File: zeppelin-server/src/test/java/org/apache/zeppelin/integration/ParagraphActionsIT.java [hunk 4]
```diff
@@ -269,4 +270,62 @@ public void testDisableParagraphRunButton() throws Exception {
...
+
+      String xpathToTile = getParagraphXPath(1) + "//div[contains(@class, 'title')]/div";
+      String xpathToSettingIcon = getParagraphXPath(1) + "//span[@class='icon-settings']";
+
+      collector.checkThat("Before Show Title : The title field contains",
+          driver.findElement(By.xpath(xpathToTile)).getText(),
+          CoreMatchers.equalTo(""));
+      driver.findElement(By.xpath(xpathToSettingIcon)).click();
+      collector.checkThat("Before Show Title : The title option in option panel of paragraph is labeled as  ",
+          driver.findElement(By.xpath(getParagraphXPath(1) + "//a[contains(@ng-click, 'showTitle()')]")).getText(),
```
Comment (corneadoug) [inline zeppelin-server/src/test/java/org/apache/zeppelin/integration/ParagraphActionsIT.java]: Isn't `//a[contains(@ng-click, 'showTitle()')]` the same as `//ul/li/a[@ng-click='showTitle()']` a bit lower in the code?

File: zeppelin-server/src/test/java/org/apache/zeppelin/integration/ParagraphActionsIT.java [hunk 5]
```diff
@@ -269,4 +270,62 @@ public void testDisableParagraphRunButton() throws Exception {
...
+    if (!endToEndTestEnabled()) {
+      return;
+    }
+    try {
+      createNewNote();
+
+      waitForParagraph(1, "READY");
+      ZeppelinITUtils.sleep(1000, false);
+
+      String xpathToTile = getParagraphXPath(1) + "//div[contains(@class, 'title')]/div";
```
Comment (corneadoug) [inline zeppelin-server/src/test/java/org/apache/zeppelin/integration/ParagraphActionsIT.java]: xpathToTile -> xpathToTitle

**Comment by felixcheung** [comment]
@ravicodder could you rebase please?

**Comment by ravicodder** [comment]
@felixcheung done rebase.

**Comment by corneadoug** [comment]
LGTM

**Pull request closed without merging**